File: backend/app/schemas/transcribe.py
from typing import List, Optional
from pydantic import BaseModel

class TranscriptSegment(BaseModel):
    segment_id: int
    start: float
    end: float
    text_ja: str

class TranscriptionResponse(BaseModel):
    full_text_ja: str
    segments: List[TranscriptSegment]
    duration: float
    media_url: Optional[str] = None
    media_kind: Optional[str] = None
    audio_filename: Optional[str] = None

File: frontend/src/lib/types/transcribe.ts
export type TranscriptSegment = {
  segment_id: number;
  start: number;
  end: number;
  text_ja: string;
};

export type TranscriptResponse = {
  audio_filename: string;
  duration: number;
  full_text_ja: string;
  segments: TranscriptSegment[];
  media_url: string;
  media_kind: "audio" | "video";
};
